sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -78,29 +78,66 @@ class SessionCatalog(
     if (conf.caseSensitiveAnalysis) name else name.toLowerCase
   }
 
+  /**
+   * Validate names
+   */
+  protected[this] def validateName(name: String): Boolean = {
+    val validName = "([_A-Za-z0-9]*)".r
```

Comment:
Thanks for catching it! Although the name `___` is weird, but it is allowable. Any reason we should stop users using it?